WolvenKit.RED4/Archive/IO/CR2WReader.cs

```diff
@@ -100,14 +100,14 @@ public bool ReadVariable(RedBaseClass cls, ExtendedTypeInfo typeInfo)
 
             var propRedType = RedReflection.GetRedTypeFromCSType(prop.Type, prop.Flags);
 
-            if (propRedType != redTypeName)
+            if (redTypeName != CName.Empty && propRedType != redTypeName)
```

Comment:
This changes serialization logic (the string check), why?

Reply:
because we can't set a property from an empty string, it'll throw exceptions.

Reply:
`redTypeName != CName.Empty` isn't a nullable check.
`CName` is a value type and can't be null (also CName.Empty isn't null either (it's 0), so doesn't make sense here). 
Whats giving the warning is the exception constructor (or `TryReadValue`) which wants a string. It tries to convert the CName to string via implicit conversion which can yield null (unresolveable string).

Adding the `!` below (as you did) should be enough, since impossible that the string can't be resolved at this point.

Same for the other below

Reply:
Thanks, Seb <3